tests/PHPStan/Analyser/nsrt/preg_match_shapes.php

```diff
@@ -768,3 +768,55 @@ function bug11604b (string $string): void {
 		assertType("array{0: string, 1?: ''|'XX', 2?: ''|'YY', 3?: 'ZZ'}", $matches);
 	}
 }
+
+function testLtrimDelimiter (string $string): void {
+	if (preg_match('/(x)/', $string, $matches)) {
+		assertType("array{string, 'x'}", $matches);
+	}
+}
+
+function testUnescapeBackslash (string $string): void {
+	if (preg_match(<<<'EOD'
```

Comment:
is this nowdoc usage important for the tests or can it be simplified?

Reply:
Of course not, but it simplied the human understanding of many backslashes which needs to be otherwise escaped in php strings...